swan-lake/development-tutorials/run-in-the-cloud/deployment-guide/ballerina-k8s-deployment-guide.md

```diff
@@ -0,0 +1,357 @@
+     $ curl http://192.168.49.2:30342/helloWorld/sayHello
+     Hello, Kubernetes!
+    ```
+    Follow the below references for more information:
+  - [Deploy Ballerina on Kubernetes](https://ballerina.io/learn/deploy-ballerina-on-kubernetes/)
+  - [Kubernetes - Hello world](https://ballerina.io/learn/by-example/kubernetes-hello-world/)
+
+## Ballerina deployment with Kubernetes Kustomize
+
+[Kustomize](https://kustomize.io/) is a tool that allows you to modify Kubernetes YAML files without altering 
+the original files. It can enhance the generated YAML from code-to-cloud deployments by applying additional 
+customizations. The `kustomization.yaml` file in the root directory serves as a sample, combining the generated YAMLs 
+from different projects. Environment variables, such as specifying the location of the `Config.toml` for a service, 
+can be added using Kustomize patches, enabling further configuration.
+
+**kustomization.yaml**
+```yaml
+resources:
+  - /path/to/target/kubernetes/hotel/hotel.ymal
+  - /path/to/target/kubernetes/room/room.yaml
+  - …
+
+patchesStrategicMerge:
+- secret-env-patch.yaml
+
+```
+See [Kustomize](https://github.com/ballerina-guides/gcp-microservices-demo?tab=readme-ov-file#kustomize) for more information.  
+
+## CI/CD with Kubernetes
+This GitHub Action workflow automates the continuous integration and continuous deployment (CI/CD) process for a Ballerina project. 
+It triggers every push to the repository, and builds the project.
+
+```yaml
+
+name: Ballerina CI/CD with Kubernetes and Ballerina Central Push
+
+on: [push]
+
+jobs:
+  build-and-deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Step 1: Checkout the Repository
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # Step 2: Build the Ballerina Project
+      - name: Ballerina Build
+        uses: ballerina-platform/ballerina-action@master
+        with:
+          args: build --cloud=k8s
+
+
+      # Step 3: Log in to Docker Registry
+      - name: Log in to Docker Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ secrets.DOCKER_REGISTRY_URL }}
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+
+      # Step 4: Push the Docker Image
+      - name: Push Docker Image
+        run: |
+          docker push ${{ secrets.DOCKER_REGISTRY_URL }}/your-docker-repo/your-app:${{ github.sha }}
+
+      # Step 5: Set up Kubectl
+      - name: Set up Kubectl
+        uses: azure/setup-kubectl@v3
+        with:
+          version: 'v1.22.0'
+
+      # Step 6: Deploy to Kubernetes
+      - name: Deploy to Kubernetes
+        env:
+          KUBE_CONFIG_DATA: ${{ secrets.KUBE_CONFIG_DATA }}
+        run: |
+          echo "${KUBE_CONFIG_DATA}" | base64 --decode > $HOME/.kube/config
+          kubectl apply -f <path to generated artifacts>
+```
+- **Step 01:** The push event can initiate a build and deployment process to the Kubernetes cluster.
```

Comment:
01, 1 not consistent between the workflow and comment.

Reply:
Updated please review : https://github.com/ballerina-platform/ballerina-dev-website/pull/9621/commits/6dd8e82249c550e1c25360082bafa85c155698e3